src/main/java/org/terasology/projsndwv/TempBeeRegistry.java

```diff
@@ -37,32 +42,62 @@
     private static HashMap<Integer, Integer> lifespans = new HashMap<>();
     private static HashMap<Integer, Long> tickspeeds = new HashMap<>();
     private static HashMap<Integer, Production> productions = new HashMap<>();
+    private static HashMap<Integer, HashMap<Integer, String>> genotypeNames = new HashMap<>();
 
     private static MersenneRandom random;
 
     private static Logger logger = LoggerFactory.getLogger(TempBeeRegistry.class);
 
-    public static TextureRegionAsset<?> getTextureRegionAssetForSpeciesAndType(int species, int type) {
+    public static EntityRef modifyItemForSpeciesAndType(EntityRef entity) {
+        if (!entity.hasComponent(BeeComponent.class)) {
+            return entity;
+        }
+
+        int species = entity.getComponent(GeneticsComponent.class).activeGenes.get(ApiarySystem.LOCUS_SPECIES);
```

Comment:
Would it be worth guarding against a case where an entity that _does_ have a `BeeComponent` may not have a `GeneticsComponent` ? That'd result in an NPE crash

Reply:
It's a bit either/or for both components.

If a GeneticsComponent is missing due to an internal flaw, this is a critical error. If it's missing due to an external circumstance, then the wrong function has been called.

I struggling to think of meaningful reasons as to why the ItemComponent would be missing.

Because of this, I think the NPEs are justified, though they're easily avoidable by just returning the input.

Reply:
Fair enough. I'm usually thinking in general with future changes in mind: At some point an additional use case may be added where it suddenly become valid to not have the second component, but old code relies on a previous assumption that's no longer valid.

It is good if you get a for-sure crash while developing. But if the new use case is rare do we end up risking ugly crashes during normal players using the game? Maybe it would be a case for more unit tests to help exercise things better.

In any case: general sentiment, not necessarily appropriate here :-)